tests/D2L.CodeStyle.Analyzers.Test/Specs/ImmutabilityAnalyzer.cs

```diff
@@ -315,4 +326,86 @@ sealed class ClassWithExceptedImmutableMemberSucceeds {
 		}
 
 	}
-}
+
+	class GenericTypeFieldTests {
+		// Generic type parameter state with mutable and immutable concrete types
+
+		[Objects.Immutable]
+		sealed class GenericClassWithImmutableState<[Objects.Immutable] T> {
+			internal readonly T m_GenericClassWithImmutableState;
+		}
+
+		[Objects.Immutable]
+		class /* ImmutableClassIsnt('m_ConcreteClassWithMutableGenericType.m_GenericClassWithImmutableState.m_MutableClass' is not read-only) */ ConcreteClassWithMutableGenericType /**/ {
+			internal readonly GenericClassWithImmutableState<MutableClass> m_ConcreteClassWithMutableGenericType;
+		}
+
+		[Objects.Immutable]
+		class ConcreteClassWithImmutableGenericType {
+			internal readonly GenericClassWithImmutableState<ImmutableClass> m_ConcreteClassWithImmutableGenericType;
+		}
+	}
+
+	class GenericTypeParameterTests {
+		// Ensures generic type parameters from an interface are examined
+
+		[Objects.Immutable]
+		interface ImmutableGenericInterface<[Objects.Immutable] T> {

```

Comment:
We should check that we complain when the generic interface has an implementation that holds onto type `T`, but we should also check that if there is no implementation holding onto type `T` then everything still passes.  Fortunately, it looks like you are covered by some existing tests for that:
`GenericClassWithoutStateIsSafe<T>`